src/klein/test/_strategies.py

```diff
@@ -185,36 +208,68 @@ def hostnames(
 
     name = u".".join(labels)
 
-    return name
+    # Filter names that are not IDNA-encodable.
+    # We try pretty hard not to generate bogus names in the first place... but
+    # catching all cases is not trivial.
+    try:
+        idna_encode(name)
+    except IDNAError:
+        assume(False)
 
+    return name
 
-_path_characters = tuple(
-    unichr(i)
-    for i in range(maxunicode) if i not in (ord(c) for c in "#/?")
-)
 
-@composite
-def path_segments(draw):  # pragma: no cover
-    # type: (DrawCallable) -> Text
+def path_characters():
+    # type: () -> str
     """
-    A strategy which generates URL path segments.
+    Returns a string containing valid URL path characters.
     """
+    global _path_characters
+
+    if _path_characters is None:
+        def chars():
+            # type: () -> Iterable[Text]
+            for i in range(maxunicode):
+                c = unichr(i)
+
+                # Exclude reserved characters
+                if c in "#/?":
+                    continue
+
+                # Exclude anything not UTF-8 compatible
```

Comment:
What?

Reply:
If a path segment is not encodable to UTF-8, hyperlink will puke.

Reply:
Sorry, that was not a helpful question.

The question is:

"What code points are not encodable to UTF-8?"